### ISSUE TITLE
version 1.14.2, shared lib is ok, but static lib raises error H5::PredType::NATIVE_UINT64 <struct at NULL>

### ISSUE THREAD

**Issue by liyqxtu** [HDFGroup/hdf5#3544]
![1](https://github.com/HDFGroup/hdf5/assets/4734699/33d8e1dc-19c9-4953-89c3-90406ee3c4d6)

**Comment by bmribler** [comment]
@liyqxtu, Could you please provide more info about your system?

**Comment by liyqxtu** [comment]
The original project was complex, so I created a new project and extracted the code related to the bug.
[ConsoleApplication1.zip](https://github.com/HDFGroup/hdf5/files/12667389/ConsoleApplication1.zip)

static MyLib myLib;  // raise error H5::PredType::NATIVE_UINT64 <struct at NULL>

int main()
{
	//MyLib myLib2;    // everything is ok
	return 0;
}

**Comment by bmribler** [comment]
@liyqxtu thanks, we'll check it out.

**Comment by liyqxtu** [comment]
is there any idea to fixed bug?

**Comment by liyqxtu** [comment]
yeah, i simplify the bug project, it seems that always crash when call static library
[ConsoleApplication1.zip](https://github.com/HDFGroup/hdf5/files/12729055/ConsoleApplication1.zip)

**Comment by bmribler** [comment]
Thanks for the simplified project.  I can see the issue and looking into it.
